### PR TITLE
pymc 5.16.1 

### PR DESCRIPTION
**Pull request by psteyer**
pymc 5.16.1 

**Destination channel:** Defaults

### Links

- [PKG-5095]
- dev_url:        https://github.com/pymc-devs/pymc/tree/v5.16.1
- conda_forge:    https://github.com/conda-forge/pymc-feedstock
- pypi:           https://pypi.org/project/pymc/5.16.1
- pypi inspector: https://inspector.pypi.io/project/pymc/5.16.1/

### Explanation of changes:

- Reverting scipy to <1.13 due to pymc import failure that was caused due to older version of arviz. In the future, it could be worth updating arviz to solve the issue. See: https://github.com/pymc-devs/pymc/issues/7236
- Adding tests, and running tests form one file like documentation mentions. Otherwise tests would take a lot of time. See: https://www.pymc.io/projects/docs/en/latest/contributing/running_the_test_suite.html

[PKG-5095]: https://anaconda.atlassian.net/browse/PKG-5095?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lorepirri** [comment]
`osx-arm64` fails with
```
                   INFO   Encountered problems while solving:                                                                                                                                                
                   INFO     - nothing provides requested gcc_osx-arm64   
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,11 +32,16 @@ requirements:
     - cloudpickle
     - fastprogress >=0.2.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
Not in https://github.com/pymc-devs/pymc/blob/v5.16.1/requirements.txt

File: recipe/meta.yaml [hunk 2]
```diff
@@ -33,11 +32,16 @@ requirements:
...
     - pandas >=0.24.0
-    - pytensor >=2.12.0,<2.13
-    - scipy >=1.4.1
+    - pytensor >=2.23,<2.24
+      # Reverting scipy to <1.13 due to pymc import failure that was caused due to older version of arviz
+    - scipy <1.13
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - scipy >=1.4.1,<1.13
```
https://github.com/pymc-devs/pymc/blob/v5.16.1/requirements.txt#L8
Although, what would it take to update arviz?

Comment (psteyer) [reply recipe/meta.yaml]: Doesn't seem to difficult looking at the recipe. Our highest version is `0.17.1` and current version I'm seeing on pypi is `0.18.0`

**Review by JeanChristopheMorinPerso — COMMENTED** [review]
The license is `APACHE-2.0 AND MIT` as highlighted in https://github.com/pymc-devs/pymc/blob/main/LICENSE. More particularly, code in https://github.com/pymc-devs/pymc/tree/main/pymc/logprob is under MIT.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by ViridianMelody — APPROVED** [review]

**Pull request merged**